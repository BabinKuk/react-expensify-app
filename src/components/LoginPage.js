import React from 'react';
import { connect} from 'react-redux';
import { startGoogleLogin, startGithubLogin } from '../actions/auth';

export const LoginPage = ({ startGoogleLogin, startGithubLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify App</h1>
            <p>It's time to get your expenses under control.</p>
            <div className="box-layout__buttons">
                {/*<button onClick={startGoogleLogin} className="button">
                    <img src=""></img>
                </button>
                <button onClick={startGithubLogin} className="button">
                    Login with Git
                </button>*/}
                <img src="./images/google-64x.png" onClick={startGoogleLogin} className="button--login" alt="Login with Google"></img>
                <img src="./images/GitHub-Mark-64px.png" onClick={startGithubLogin} className="button--login" alt="Login with Github"></img>
            </div>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startGoogleLogin: () => dispatch(startGoogleLogin()),
    startGithubLogin: () => dispatch(startGithubLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);

