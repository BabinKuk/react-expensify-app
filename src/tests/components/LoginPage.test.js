import React from 'react';
import { shallow } from 'enzyme';
import LoginPage from '../../components/LoginPage';

test('Test LoginPage rendering', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('Test LoginPage startLogin', () => {
    // spy function
    const startLogin = jest.fn();

    const wrapper = shallow(<LoginPage startLogin={startLogin} />);

    //simulate click event
    wrapper.find('button').simulate('click');

    expect(startLogin).toHaveBeenCalled();
});

