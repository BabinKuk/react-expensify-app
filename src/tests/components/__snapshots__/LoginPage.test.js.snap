// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LoginPage rendering 1`] = `
<div>
  <button>
    Login
  </button>
</div>
`;
